plugins/module_utils/_global_catalog.py

```diff
@@ -57,132 +68,64 @@ def _get_catalog_records(stdout):  # type: (str) -> tuple[str | None, str | None
-    """
-    Validates that the JCL line does not exceed MAX_LINE_LENGTH.
-    Raises ValueError if validation fails.
-    """
-    if len(line) > MAX_LINE_LENGTH:
-        raise ValueError(f"Value supplied for {parameter} ({name}) is {len(line) - MAX_LINE_LENGTH} characters too long")
-    return True
-
-
 def _run_dfhrmutl(
         location,  # type: str
         sdfhload,  # type: str
         cmd=""  # type: str
 ):
     # type: (...) -> tuple[list[dict[str, str| int]], tuple[str | None, str | None]] | list[dict[str, str| int]]
-
-    qualified_file_path = _create_dfhrmutl_jcl(
-        location,
-        sdfhload,
-        cmd
-    )
     executions = []
-    try:
-        for x in range(MVS_CMD_RETRY_ATTEMPTS):
-            dfhrmutl_response, jcl_executions = _execute_dfhrmutl(qualified_file_path)
-            dfhrmutl_rc = dfhrmutl_response["ret_code"].get("code")
-
-            allContent = []
-            for ddname in dfhrmutl_response.get("ddnames"):
-                allContent += ddname.get("content")
-            stdout_raw = "".join(allContent)
-
-            executions.append(jcl_executions)
-            executions.append(
-                _execution(
-                    name="DFHRMUTL - {0} - Run {1}".format(
-                        "Get current catalog" if cmd == "" else "Updating autostart override",
-                        x + 1),
-                    rc=dfhrmutl_rc,
-                    stdout=stdout_raw,
-                    stderr=dfhrmutl_response["ret_code"].get("msg_txt", "")))
-
-            if dfhrmutl_rc not in (0, 16):
+
+    for x in range(MVS_CMD_RETRY_ATTEMPTS):
+        dfhrmutl_response = _execute_dfhrmutl(location, sdfhload, cmd)
+        executions.append(
+            _execution(
+                name="DFHRMUTL - {0} - Run {1}".format(
+                    "Get current catalog" if cmd == "" else "Updating autostart override",
+                    x + 1),
+                rc=dfhrmutl_response.rc,
+                stdout=dfhrmutl_response.stdout,
+                stderr=dfhrmutl_response.stderr))
+
+        if dfhrmutl_response.rc == 0:
+            break
+        if dfhrmutl_response.rc == 16:
+            formatted_stdout_lines = [
+                "{0}".format(element.replace(" ", "").upper())
+                for element in dfhrmutl_response.stdout.split("\n")
+            ]
+            stdout_with_rc = list(filter(lambda x: "REASON:X" in x, formatted_stdout_lines))
+
+            reason_code = _get_reason_code(stdout_with_rc)
+            if reason_code and reason_code != "A8":
+                raise MVSExecutionException(
+                    "DFHRMUTL failed with RC 16 - {0}".format(stdout_with_rc[0]), executions
+                )
+            elif reason_code is None:
+                raise MVSExecutionException(
+                    "DFHRMUTL failed with RC 16 but no reason code was found",
+                    executions,
+                )
+        else:
```

Comment:
Not sure this if is right? When DFHRMUTL passes (with ZOAU 1.3) but the mvscmd wrapper fails, we get an rc 16 from mvscmd so we would fall into the if when really we want to check the stderr for the child process return code (DFHRMUTLs actual return code) and stdout for the program header to make sure it was the right program that ran.